src/components/EmailForm.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import Form from "./reusables/Form";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import { FetchErrorData } from "opds-web-client/lib/interfaces";
+import { Store } from "redux";
+import { State } from "../reducers/index";
+import { LibraryData } from "../interfaces";
+import { CheckSoloIcon } from "@nypl/dgx-svg-icons";
+
+export interface EmailFormState {
+  sent: boolean;
+}
+
+export interface EmailFormOwnProps {
+  library: LibraryData;
+  store: Store<State>;
+}
+
+export interface EmailFormDispatchProps {
+  email: (data: FormData) => Promise<void>;
+}
+
+export interface EmailFormStateProps {
+  error?: FetchErrorData;
+}
+
+export interface EmailFormProps extends EmailFormOwnProps, EmailFormDispatchProps, EmailFormStateProps {}
+
+export class EmailForm extends React.Component<EmailFormProps, EmailFormState> {
+
+  constructor(props) {
+    super(props);
+    this.sendEmail = this.sendEmail.bind(this);
+    this.state = { sent: false };
+  }
+
+  async sendEmail(data: FormData): Promise<void> {
+    await this.props.email(data);
+    this.setState({ sent: true });
+  }
+
+  render(): JSX.Element {
+    let disabled = false;
+    let icon = this.state.sent ? <CheckSoloIcon iconId="1" /> : null;
```

Comment:
The `CheckSoloIcon` doesn't use `iconId` but it's good you tried to add it. If it was working, you shouldn't use numbers for ids like this. Also, screenreaders will give you a "duplicate id on page" error if they read multiple elements with the same id. In this case, I think it'll happen since you would have X number of validation forms for X number of libraries.